src/ast/CodeGeneratorEvaluator.ipp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <cstddef>
+
+#include "nodes.hpp"
+#include "scopes/scopeStack.hpp"
+
+namespace ast
+{
+
+static constexpr char ENDL = '\n';
+static constexpr char INDENT = '\t';
+
+class CodeGeneratorEvaluator
```

Comment:
The naming is not correct: it's not an evaluator, it doesn't implement operator()